docs/manual/other-topics/typescript.md

```diff
@@ -16,131 +17,177 @@ In order to avoid installation bloat for non TS users, you must install the foll
 
 ## Usage
 
-Example of a minimal TypeScript project with strict type-checking for attributes.
-
 **Important**: You must use `declare` on your class properties typings to ensure TypeScript does not emit those class properties.
 See [Caveat with Public Class Fields](./model-basics.html#caveat-with-public-class-fields)
 
-**NOTE:** Keep the following code in sync with `/types/test/typescriptDocs/ModelInit.ts` to ensure it typechecks correctly.
+Sequelize Models accept two generic types to define what the model's Attributes & Creation Attributes are like:
 
 ```typescript
-/**
- * Keep this file in sync with the code in the "Usage" section in typescript.md
- */
-import {
-  Association, DataTypes, HasManyAddAssociationMixin, HasManyCountAssociationsMixin,
-  HasManyCreateAssociationMixin, HasManyGetAssociationsMixin, HasManyHasAssociationMixin, Model,
-  ModelDefined, Optional, Sequelize
-} from "sequelize";
+import { Model, Optional } from 'sequelize';
 
-const sequelize = new Sequelize("mysql://root:asd123@localhost:3306/mydb");
+// We don't recommend doing this. Read on for the new way of declaring Model typings.
 
-// These are all the attributes in the User model
-interface UserAttributes {
-  id: number;
-  name: string;
-  preferredName: string | null;
+type UserAttributes = {
+  id: number,
+  name: string,
+  // other attributes...
+};
+
+// we're telling the Model that 'id' is optional
+// when creating an instance of the model (such as using Model.create()).
+type UserCreationAttributes = Optional<UserAttributes, 'id'>;
+
+class User extends Model<UserAttributes, UserCreationAttributes> {
+  declare id: number;
+  declare string: number;
+  // other attributes...
 }
+```
 
-// Some attributes are optional in `User.build` and `User.create` calls
-interface UserCreationAttributes extends Optional<UserAttributes, "id"> {}
+This solution is verbose. Sequelize v6 provides new utility types that will drastically reduce the amount
```

Comment:
Maybe include which version of v6 adds this?

Reply:
I don't know which version of v6 it is yet but we can do this just before a merge :)